manifests/config.pp

```diff
@@ -6,14 +6,24 @@
 
   assert_private()
 
-  file { $::telegraf::config_file:
-    ensure  => file,
-    content => template('telegraf/telegraf.conf.erb'),
-    mode    => '0640',
-    owner   => 'telegraf',
-    group   => 'telegraf',
-    notify  => Class['::telegraf::service'],
-    require => Class['::telegraf::install'],
+  file {
```

Comment:
Nice, I was wondering when some Puppet 4 syntax would finally creep in ;)

This is great, but you'll need to update the module's metadata to state this as a requirement.  I'll then call it out as a breaking change in version 2.0.